src/shared/intsafe.rs

```diff
@@ -4,3 +4,4624 @@
+        S_OK
+    } else {
+        *plResult = LONG_ERROR;
+        INTSAFE_E_ARITHMETIC_OVERFLOW
+    }
+}
+#[cfg(target_arch = "x86_64")] #[inline]
+pub fn ULongToLongPtr(ulOperand: ULONG, plResult: &mut LONG_PTR) -> HRESULT {
+    *plResult = ulOperand as LONG_PTR;
+    S_OK
+}
+#[cfg(target_arch = "x86")] #[inline]
+pub fn ULongToLongPtr(ulOperand: ULONG, plResult: &mut LONG_PTR) -> HRESULT {
+    ULongToLong(ulOperand, plResult)
+}
+#[inline]
+pub fn ULongToPtrdiffT(ulOperand: ULONG, pResult: &mut ptrdiff_t) -> HRESULT {
+    ULongToIntPtr(ulOperand, pResult)
+}
+#[inline]
+pub fn ULongToSSIZET(ulOperand: ULONG, pResult: &mut SSIZE_T) -> HRESULT {
+    ULongToLongPtr(ulOperand, pResult)
+}
+#[inline]
+pub fn ULongPtrToInt8(ulOperand: ULONG_PTR, pi8Result: &mut INT8) -> HRESULT {
+    if ulOperand <= INT8_MAX as ULONG_PTR {
+        *pi8Result = ulOperand as INT8;
+        S_OK
+    } else {
+        *pi8Result = INT8_ERROR;
+        INTSAFE_E_ARITHMETIC_OVERFLOW
+    }
+}
+#[inline]
+pub fn ULongPtrToUChar(ulOperand: ULONG_PTR, pch: &mut UCHAR) -> HRESULT {
+    if ulOperand <= 255 {
+        *pch = ulOperand as UCHAR;
+        S_OK
+    } else {
+        *pch = 0;
+        INTSAFE_E_ARITHMETIC_OVERFLOW
+    }
+}
+#[inline]
+pub fn ULongPtrToChar(ulOperand: ULONG_PTR, pch: &mut CHAR) -> HRESULT {
+    ULongPtrToInt8(ulOperand, pch)
+}
+#[inline]
+pub fn ULongPtrToUInt8(ulOperand: ULONG_PTR, pui8Result: &mut UINT8) -> HRESULT {
+    if ulOperand <= UINT8_MAX as ULONG_PTR {
+        *pui8Result = ulOperand as UINT8;
+        S_OK
+    } else {
+        *pui8Result = UINT8_ERROR;
+        INTSAFE_E_ARITHMETIC_OVERFLOW
+    }
+}
+#[inline]
+pub fn ULongPtrToByte(ulOperand: ULONG_PTR, pui8Result: &mut BYTE) -> HRESULT {
+    ULongPtrToUInt8(ulOperand, pui8Result)
+}
+#[inline]
+pub fn ULongPtrToShort(ulOperand: ULONG_PTR, psResult: &mut SHORT) -> HRESULT {
+    if ulOperand <= SHORT_MAX as ULONG_PTR {
+        *psResult = ulOperand as SHORT;
+        S_OK
+    } else {
+        *psResult = SHORT_ERROR;
+        INTSAFE_E_ARITHMETIC_OVERFLOW
+    }
+}
+#[inline]
+pub fn ULongPtrToInt16(ulOperand: ULONG_PTR, pi16Result: &mut INT16) -> HRESULT {
+    ULongPtrToShort(ulOperand, pi16Result)
+}
+#[inline]
+pub fn ULongPtrToUShort(ulOperand: ULONG_PTR, pusResult: &mut USHORT) -> HRESULT {
+    if ulOperand <= USHORT_MAX as ULONG_PTR {
+        *pusResult = ulOperand as USHORT;
+        S_OK
+    } else {
+        *pusResult = USHORT_ERROR;
+        INTSAFE_E_ARITHMETIC_OVERFLOW
+    }
+}
+#[inline]
+pub fn ULongPtrToUInt16(ulOperand: ULONG_PTR, pui16Result: &mut UINT16) -> HRESULT {
+    ULongPtrToUShort(ulOperand, pui16Result)
+}
+#[inline]
+pub fn ULongPtrToWord(ulOperand: ULONG_PTR, pwResult: &mut WORD) -> HRESULT {
+    ULongPtrToUShort(ulOperand, pwResult)
+}
+#[inline]
+pub fn ULongPtrToInt(ulOperand: ULONG_PTR, piResult: &mut INT) -> HRESULT {
+    if ulOperand <= INT_MAX as ULONG_PTR {
+        *piResult = ulOper
```

Comment:
If the signature including the return type does not fit on one line, then it has to be block indented with one parameter per line.